samples/cxx/openmp_ignition/openmp_ignition.cpp

```diff
@@ -19,17 +19,18 @@ void run()
     // Containers for Cantera objects to be used in different. Each thread needs
     // to have its own set of linked Cantera objects. Multiple threads accessing
     // the same objects at the same time will cause errors.
-    std::vector<std::unique_ptr<IdealGasMix>> gases;
+    std::vector<std::shared_ptr<Solution>> sols;
     std::vector<std::unique_ptr<IdealGasConstPressureReactor>> reactors;
     std::vector<std::unique_ptr<ReactorNet>> nets;
 
     // Create and link the Cantera objects for each thread. This step should be
     // done in serial
     for (int i = 0; i < nThreads; i++) {
-        gases.emplace_back(new IdealGasMix("gri30.xml", "gri30"));
+        auto sol = newSolution("gri30.yaml", "gri30", "None");
+        sols.emplace_back(sol);
```

Comment:
Why not skip the local variable and just write `sols.emplace_back(newSolution("gri30.yaml", "gri30", "None"));`?

Reply:
it's used again for `reactors.back()->insert(sol);` ... going with readability here.